Update reviewers-maintainers with new role information. [RELATED #61] (#80)
o3de/sig-docs-community: governance/reviewers-maintainers.md

## Documentation and Community reviewer roles

The Documentation and Community SIG maintains the following reviewer teams:

* [**Documentation Reviewer**](https://github.com/orgs/o3de/teams/docs-reviewers) - Responsible for reviewing documentation content to `o3de.org`, or conducting reviews of certain `o3de` in-product documentation elements.
* **Community Reviewer** - Responsible for reviewing community materials, including contributor guides and blogs.

> We also have [SIG maintainers](https://github.com/o3de/o3de/discussions). Documentation and Community maintainers are reviewers who have been elevated from the documentation or community reviewer pools to take on additional responsibility, and serve as trusted sources of guidance.

Reviewer: ```suggestion
We also have [SIG maintainers](https://github.com/o3de/o3de/discussions). Documentation and Community maintainers are reviewers who have been elevated from the documentation or community reviewer teams to take on additional responsibility and serve as trusted sources of guidance.
```

### Technical reviewer roles

> The Documentation and Community SIG also maintains a documentation technical reviewer team for the `o3de.org` website and its related infrastructure. Documentation technical reviewers are responsible for representing their SIG during documentation review, and ensuring technical accuracy of content.

Author: Tried to make this clearer about the tech reviewer role and removed the link for now. If we automate the system of review assignment, it might make more sense to have each SIG maintain its own independent reviewer group rather than contribute to a global pool.

Reviewer: Great point! I agree with this. I want to be cautious about automating review assignment for other SIG's technical review teams because they may want to handle that differently.

## Documentation and Community reviewer roles

> The Documentation and Community SIG maintains the following reviewer pools:

Reviewer: I suggest using the term "teams", so it more easily correlates to the idea of GitHub teams, if anyone was trying to understand that community architecture. 
```suggestion
The Documentation and Community SIG maintains the following reviewer teams:
```

> * **Documentation Reviewer** - Responsible for reviewing documentation content to `o3de.org`, or conducting reviews of certain `o3de` in-product documentation elements.

Reviewer: - Add link to team
- I'm not entirely sure what "in-product documentation elements" means. Perhaps you can give a simple example? 
```suggestion
* [**Documentation Reviewer**](https://github.com/orgs/o3de/teams/docs-reviewers) - Responsible for reviewing documentation content to `o3de.org`, or conducting reviews of certain `o3de` in-product documentation elements.
```

> * **Community Reviewer** - Responsible for reviewing community materials, including contributor guides and blogs.

Reviewer: I suggest adding a link to the teams, when you have one.

Author: I think each PR includes a change linking to the template for nominations in that role.

> We also have maintainers, as other SIGs do. Documentation and Community maintainers are reviewers who have been elevated from the documentation or community

Reviewer: Again, suggest adding a link. As well as for Community maintainers when you have one. 
```suggestion
We also have maintainers, as other SIGs do. [Documentation](https://github.com/orgs/o3de/teams/docs-maintainers) and Community maintainers are reviewers who have been elevated from the documentation or community.
```

reviewer pools to take on additional responsibility, and serve as trusted sources of guidance.

### Technical reviewer roles

> The Documentation and Community SIG also maintains a technical reviewer pool for the `o3de.org` website and its related infrastructure.

Reviewer: ```suggestion
The Documentation and Community SIG also maintains a [technical reviewer](https://github.com/orgs/o3de/teams/docs-tech-reviewers) team for the `o3de.org` website and its related infrastructure.
```

> The technical SIGs also maintain reviewer pools for becoming a technical reviewer. 

Reviewer: I think it would be helpful to differentiate this kind of a technical reviewer vs. a docs technical reviewer. You can link to the community's guidelines on the roles: https://github.com/o3de/community/blob/main/community-membership.md
